packages/collection-model/lib/model.js

```diff
@@ -80,26 +251,44 @@ const CollectionCollection = AmpersandCollection.extend({
         );
       }
 
-      dataService.listCollectionsNamesOnly(databaseName, (err, collections) => {
-        if (err) {
-          reject(err);
-          return;
+      let collections = [];
+
+      // When trying to fetch additional information about collections during
+      // collection list fetch we want to fallback to the nameOnly method that
+      // requires less privileges in case user is missing some required ones
+      if (fetchInfo) {
+        try {
+          collections = await listCollectionsAsync(databaseName, {});
+        } catch (e) {
```

Comment:
I guess this os more a general codebase-wide style concern than something related to this specifically, but shouldn't this only swallow specific errors? Right now any error of any kind will cause it to fallback.

Reply:
I think most of the possible server errors that can be returned here we want to ignore and try to fallback to the `listCollectionsNamesOnly` just in case, I honestly would struggle to find all the possible ones returned from this particular server command, I don't think that any documentation specifically about that exists tbh. Are you thinking more about a case where our code is broken to the point that this throws something or do you think we should track down possible server/driver errors and match against them before proceeding?

Reply:
Is there some kind of common pattern on server errors we could check to see if it exists? Like a `code` property perhaps.

My thinking is that even a `Uncaught TypeError: Cannot read property 'foo' of undefined` will land here.

Reply:
Yeah, this makes sense and we probably can rely on `code` being there for that check, I'll dig into driver doc to confirm

Reply:
I wouldn't be too surprised if there's nothing on database errors to differentiate them from other errors in which case I guess you might as well leave it. Alternatively we could annotate db errors in data service or something where we can at least tell as close to the db as possible that it is a db error.

But even then if we had a bug like we're sending a query to the db that's invalid it will also just fall back without us knowing.

It might just be a wider problem we can't fix in this PR, though :)